test/cql-pytest/test_permissions.py

```diff
@@ -553,3 +554,174 @@ def test_native_functions_always_exeutable(cql):
+# being accessible. The third test is about newly created CDC being accessible.
+def test_auto_grant_base(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key"
+            with new_test_table(user_session, test_keyspace, schema) as table:
+                # Check that the username was auto-granted SELECT permissions
+                user_session.execute(f'SELECT * FROM {table}')
+                # Check that the username was auto-granted MODIFY permissions
+                user_session.execute(f'INSERT INTO {table}(p) VALUES(42)')
+                # Check that the username was auto-granted ALTER permissions
+                user_session.execute(f"ALTER TABLE {table} WITH comment = 'hey'")
+                # When this scope ends, the test table is dropped by
+                # user_session, so we check that DROP permissions were auto-
+                # granted as well.
+
+# Test that a role creating a materialized view is automatically granted
+# permissions to read it.
+def test_auto_grant_view(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key, v int"
+            with new_test_table(user_session, test_keyspace, schema) as table:
+                # Because username has CREATE permissions, it can also use
+                # CREATE MATERIALIZED VIEW
+                mv = f'{test_keyspace}.{unique_name()}'
+                user_session.execute(f'CREATE MATERIALIZED VIEW {mv} AS SELECT * FROM {table} WHERE p IS NOT NULL AND v IS NOT NULL PRIMARY KEY (v, p)')
+                # Check that the username was auto-granted SELECT permissions
+                # on the newly created materialized view:
+                user_session.execute(f'SELECT * FROM {mv}')
+                # Check that the username was auto-granted ALTER permissions
+                # on the materialized view
+                user_session.execute(f"ALTER MATERIALIZED VIEW {mv} WITH comment = 'hey'")
+                # Check that the username was auto-granted DROP permissions
+                # on the materialized view
+                user_session.execute(f'DROP MATERIALIZED VIEW {mv}')
+
+# Test that a role creating CDC is automatically granted permissions to access
+# the CDC log table.
+# This test has two variants - cdc_on_create=True creates the table immediately
+# with CDC enabled, while cdc_on_create=True creates a table first and only
+# then enables CDC. Both should work.
```

Comment:
Nit: cdc_on_create=True appears twice in this comment. The second one was supposed to be cdc_on_create=False, right?

Reply:
Fixed.